src/components/TodoList/TodoList.tsx

```diff
@@ -0,0 +1,139 @@
+/* eslint-disable jsx-a11y/label-has-associated-control */
+
+import { useEffect, useRef } from 'react';
+import { Todo } from '../../types/Todo';
+import cn from 'classNames';
+
+interface Props {
+  todos: Todo[];
+  tempTodo: Todo | null;
+  selectedTodoId: number[];
+  isEditing: number | null;
+  newTitle: string;
+  onDelete: (value: number) => void;
+  onToggle: (value: number, index: number) => void;
+  onStartEditing: (value: string, id: number) => void;
+  onNewTodoTitle: (value: string) => void;
+  onChangeTodoTitle: (
+    event: React.FormEvent,
+    todo: Todo,
+    index: number,
+  ) => void;
+  onKey: (event: React.KeyboardEvent) => void;
+}
+
+export const TodoList: React.FC<Props> = ({
+  todos,
+  tempTodo,
+  selectedTodoId,
+  isEditing,
+  newTitle,
+  onDelete,
+  onToggle,
+  onStartEditing,
+  onNewTodoTitle,
+  onChangeTodoTitle,
+  onKey,
+}) => {
+  const newTitleFiled = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    const currentField = newTitleFiled.current;
+
+    if (currentField !== null) {
+      currentField.focus();
+    }
+  }, [newTitleFiled, isEditing, todos]);
+
+  return (
+    <section className="todoapp__main" data-cy="TodoList">
+      {todos.map((todo, i) => (
```

Comment:
Do you need index here
```suggestion
      {todos.map((todo) => (
```